rkt/pods.go

```diff
@@ -152,43 +162,41 @@ func walkPods(include includeMask, f func(*pod)) error {
 // newPod creates a new pod directory in the "preparing" state, allocating a unique uuid for it in the process.
 // The returned pod is always left in an exclusively locked state (preparing is locked in the prepared directory)
 // The pod must be closed using pod.Close()
-func newPod() (*pod, error) {
+func newPod() (p *pod, err error) {
 	if err := initPods(); err != nil {
 		return nil, err
 	}
 
-	p := &pod{
+	p = &pod{
 		createdByMe: true,
 		isEmbryo:    true, // starts as an embryo, then xToPreparing locks, renames, and sets isPreparing
 		// rest start false.
 	}
 
-	var err error
 	p.uuid, err = types.NewUUID(uuid.New())
 	if err != nil {
 		return nil, errwrap.Wrap(errors.New("error creating UUID"), err)
 	}
 
 	err = os.Mkdir(p.embryoPath(), 0750)
 	if err != nil {
-		return nil, err
+		return
```

Comment:
this could return both `p != nil` and an error. It is cleaner to return either a pod or an error but not both.

Ditto below.